metricflow/data_table/mf_column.py

```diff
@@ -0,0 +1,39 @@
+from __future__ import annotations
+
+from dataclasses import dataclass
+from functools import cached_property
+from typing import Iterator, Sequence, Tuple, Type
+
+from metricflow.data_table.column_types import CellValue
+
+
+@dataclass(frozen=True)
+class ColumnDescription:
+    """Describes a single column in a data table."""
+
+    column_name: str
+    column_type: Type[CellValue]
+
+    def lower_case_column_name(self) -> ColumnDescription:  # noqa: D102
```

Comment:
nit: given that method name I'd expect this to be a string. Given how it's currently used I think this is better:

```
def with_lower_case_column_name(self) -> ColumnDescription:
```

Reply:
Updated.